Source/WebCore/platform/graphics/PlatformTimeRanges.h

```diff
@@ -101,7 +101,7 @@ class PlatformTimeRanges {
 
         inline bool isContiguousWithRange(const Range& range) const
         {
-            return range.m_start == m_end || range.m_end == m_start;
+            return range.m_start.compare(m_end, true) == MediaTime::EqualTo || range.m_end.compare(m_start, true) == MediaTime::EqualTo;
```

Comment:
There are micro-discontinuities (<0.00002 sec.) in the ranges reported by qtdemux in the Youtube2015 tests (don't know if the problem is in the video file or in qtdemux). These fuzzy comparisons help to overcome that problem. Without them, instead of having the buffered ranges as a a big growing range, we have lots of tiny discontinuous ranges that cause all kinds of problems in the tests.